proxy/src/cancellation.rs

```diff
@@ -107,6 +136,15 @@ impl<P: CancellationPublisher> CancellationHandler<P> {
             }
             return Ok(());
         };
+
+        if let Some(addr) = peer_addr {
+            if !addr.is_unspecified()
```

Comment:
At this point in the code the addr should be trustable. If you believe we should check for unspecified, it should be done much sooner when we create ctx.